flink-runtime/build.gradle

```diff
@@ -42,6 +42,8 @@ project(':iceberg-flink-runtime') {
     implementation(project(':iceberg-nessie')) {
       exclude group: 'com.google.code.findbugs', module: 'jsr305'
     }
+
+    implementation "org.apache.flink:flink-connector-base"
```

Comment:
added comment

Reply:
@stevenzwu  After we got this PR merged: https://github.com/apache/iceberg/pull/3364,  we don't need to introduce a common `iceberg-flink-runtime` for all different flink versions, instead we have a different `iceberg-flink:iceberg-flink-<MAJOR.MINO>-runtime` module for different `<MAJRO.MINOR>` flink releases so that we could build the features on top of the latest flink API. 

You may want to add the transitive dependency `org.apache.flink:flink-connector-base` in[ this line for flink 1.12](https://github.com/apache/iceberg/blob/master/flink/v1.12/build.gradle#L154) , and[ this line for flink 1.13.](https://github.com/apache/iceberg/blob/master/flink/v1.13/build.gradle#L154)